kustomize/components/memorystore/kustomization.yaml

```
@@ -28,7 +28,7 @@ patchesStrategicMerge:
           - name: server
             env:
             - name: REDIS_ADDR
-              value: "{{REDIS_ADDR}}"
+              value: "REDIS_ADDR"
```

Question: Is there a reason we remove the curly braces? 
Should we also fix:
```
sed -i "s/{{REDIS_ADDR}}/${REDIS_IP}:${REDIS_PORT}/g" components/memorystore/kustomization.yaml
```
inside `kustomize/components/memorystore/README.md`?

---

Maybe it was an accident?
```suggestion
              value: "{{REDIS_ADDR}}"
```

---

Hey Nim, good question, actually I put that in the description: `Review the setup of tokens to avoid issues with Kustomize and bash session`. So that's on purpose otherwise we have issue in bash session when doing the `sed` part and/or when doing `kustomize build .` without these tokens replaced. Let me know if you need more info. But the tl,dr is we should have token just like this `REDIS_ADDR` not with any `{}` nor `{{}}`, etc.

---

Following this today, one possible unintended side effect is that the `sed` now matches the name and value. So I end up with:

```
            - name: 10.34.24.75:6379
              value: "10.34.24.75:6379"
```

---

Yep, that's why we document this like this: https://github.com/GoogleCloudPlatform/microservices-demo/pull/1102/commits/f60a67fabbd855df4c0f703ea6bb3918e4b214f1. But yeah if someone is not doing the `\"` it could land to error. Open for suggestions to improve this for the different files having tokens like this. PR about that are more than welcome ;)

---

Note: utlimately, I'm wondering if we have something to do with Kustomize's [`replacements`](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/) (knowing that [`vars`](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/vars/) are planned to be deprecated).

---

Interesting. I didn't run the command manually, I was following the [terraform](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/terraform) example when that happened. I'll try running it again to see if I get the same result?

---

Yes please, if you run into any issue, feel free to file an Issue/PR for that. For Terraform, there is this https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/terraform/memorystore.tf#L37, JFYI. Which is not working with what you just said?

---

Correct. I checked that I pulled latest and just reran the terraform deploy. I see the escape characters and the quote `\"` and it still seems to be replacing all of them.

To clarify, it worked in the sense that the application deployed beautifully, it just replaced all instances of `REDIS_ADDR` with `10.34.24.75:6379`. (Including the instance at the top of the `/microservices-demo/kustomize/components/memorystore/kustomization.yaml` file.)

```diff
-# cartservice - replace REDIS_ADDR to target new Memorystore (redis) instance
+# cartservice - replace 10.34.24.75:6379 to target new Memorystore (redis) instance
 - |-
   apiVersion: apps/v1
   kind: Deployment
@@ -27,8 +27,8 @@ patchesStrategicMerge:
         containers:
           - name: server
             env:
-            - name: REDIS_ADDR
-              value: "REDIS_ADDR"
+            - name: 10.34.24.75:6379
+              value: "10.34.24.75:6379"
 # redis - remove the redis-cart Deployment
 - |-
   apiVersion: apps/v1
```

---

Gotcha, thanks. Do you mind reporting this as a dedicated issue please? We will need to tackle this to fix this broken Terraform deployment with Memorystore. Thanks!

---

I'll do that!

---

FYI: this will be fixed via https://github.com/GoogleCloudPlatform/microservices-demo/pull/1202. Thanks!

---

Nice!

I also had a solution (all the backslashes!!) that I was going to PR today, but that is much cleaner: https://github.com/LukeSchlangen/microservices-demo/commit/4f1b278111782cfcdda4224d77fc501d10ada8ed